src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,154 @@
+    private static final Logger log = LoggerFactory.getLogger(SchedulerState.class);
+
+    private static final String SUPPRESSED_KEY = "suppressed";
+
+    public SchedulerState(String frameworkName, String mesosZkURI) {
+        super(frameworkName, mesosZkURI);
+    }
+
+    public FrameworkID getFrameworkId() {
+        try {
+            return fetchFrameworkId().orElse(null);
+        } catch (StateStoreException ex) {
+            log.warn("Failed to get FrameworkID. "
+                    + "This is expected when the service is starting for the first time.", ex);
+        }
+        return null;
+    }
+
+    public void setFrameworkId(FrameworkID fwkId) throws StateStoreException {
+        try {
+            log.info(String.format("Storing framework id: %s", fwkId));
+            storeFrameworkId(fwkId);
+        } catch (StateStoreException ex) {
+            log.error("Failed to set FrameworkID: " + fwkId, ex);
+            throw ex;
+        }
+    }
+
+    public void recordTasks(List<TaskInfo> taskInfos) throws StateStoreException {
+        log.info(String.format("Recording %d updated TaskInfos/TaskStatuses:", taskInfos.size()));
+        List<TaskStatus> taskStatuses = new ArrayList<>();
+        for (TaskInfo taskInfo : taskInfos) {
+            TaskStatus taskStatus = TaskStatus.newBuilder()
+                    .setTaskId(taskInfo.getTaskId())
+                    .setExecutorId(taskInfo.getExecutor().getExecutorId())
+                    .setState(TaskState.TASK_STAGING)
+                    .build();
+            log.info(String.format("- %s => %s", taskInfo, taskStatus));
+            log.info("Marking stopped task as failed: {}", TextFormat.shortDebugString(taskInfo));
+            taskStatuses.add(taskStatus);
+        }
+
+        storeTasks(taskInfos);
+        for (TaskStatus taskStatus : taskStatuses) {
+            recordTaskStatus(taskStatus);
+        }
+    }
+
+    public void updateStatus(TaskStatus taskStatus) throws StateStoreException {
+        log.info(String.format("Recording updated TaskStatus to state store: %s", taskStatus));
+        recordTaskStatus(taskStatus);
+    }
+
+    public List<TaskInfo> getTerminatedTaskInfos() throws Exception {
+        return new ArrayList(fetchTerminatedTasks());
+    }
+
+    public int getRunningBrokersCount() throws StateStoreException {
+        int count = 0;
+
+        for (TaskStatus taskStatus : getTaskStatuses()) {
+            if (taskStatus.getState().equals(TaskState.TASK_RUNNING)) {
+                count++;
+            }
+        }
+
+        return count;
+    }
+
+    @Override
+    public Set<TaskStatus> getTaskStatuses() throws StateStoreException {
+        Set<TaskStatus> taskStatuses = new HashSet<TaskStatus>();
+        taskStatuses.addAll(fetchStatuses());
+        return taskStatuses;
+    }
+
+    public List<TaskInfo> getTaskInfos() throws StateStoreException {
```

Comment:
I just copy+pasted this from the previous `FrameworkState` class in kafka.  I can remove it and we can just use `Collection` if you like.